# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/cs_blur.glsl
# Commit: 5589923a52c8
# Full Hash: 5589923a52c80ed09c8ce383afb0385cabc0ec18
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-15 09:29:54
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r 2d192d43277b -r 5589923a52c8 gfx/wr/webrender/res/cs_blur.glsl
--- a/gfx/wr/webrender/res/cs_blur.glsl	Tue Dec 15 00:01:19 2020 +0000
+++ b/gfx/wr/webrender/res/cs_blur.glsl	Tue Dec 15 00:43:28 2020 +0000
@@ -2,10 +2,11 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#define WR_FEATURE_TEXTURE_2D
+
 #include shared,prim_shared
 
 varying vec2 vUv;
-flat varying float vUvLayer;
 flat varying vec4 vUvRect;
 flat varying vec2 vOffsetScale;
 // The number of pixels on each end that we apply the blur filter over.
@@ -73,7 +74,6 @@
     RectWithSize target_rect = blur_task.common_data.task_rect;
 
     vec2 texture_size = vec2(textureSize(sColor0, 0).xy);
-    vUvLayer = src_task.texture_layer_index;
 
     // Ensure that the support is an even number of pixels to simplify the
     // fragment shader logic.
@@ -118,10 +118,10 @@
 
 #if defined WR_FEATURE_COLOR_TARGET
 #define SAMPLE_TYPE vec4
-#define SAMPLE_TEXTURE(uv)  texture(sColor0, vec3(uv, vUvLayer))
+#define SAMPLE_TEXTURE(uv)  texture(sColor0, uv)
 #else
 #define SAMPLE_TYPE float
-#define SAMPLE_TEXTURE(uv)  texture(sColor0, vec3(uv, vUvLayer)).r
+#define SAMPLE_TEXTURE(uv)  texture(sColor0, uv).r
 #endif
 
 // TODO(gw): Write a fast path blur that handles smaller blur radii
@@ -182,9 +182,8 @@
         return;
     }
 
-    int layer = swgl_textureLayerOffset(sColor0, vUvLayer);
     swgl_commitGaussianBlurRGBA8(sColor0, vUv, vUvRect, vOffsetScale.x != 0.0,
-                                 vSupport, vGaussCoefficients, layer);
+                                 vSupport, vGaussCoefficients, 0);
 }
     #else
 void swgl_drawSpanR8() {
@@ -192,9 +191,8 @@
         return;
     }
 
-    int layer = swgl_textureLayerOffset(sColor0, vUvLayer);
     swgl_commitGaussianBlurR8(sColor0, vUv, vUvRect, vOffsetScale.x != 0.0,
-                              vSupport, vGaussCoefficients, layer);
+                              vSupport, vGaussCoefficients, 0);
 }
     #endif
 #endif